### PR TITLE
Fix Tasks/Notes created with null position

### LINKED ISSUE #5268: Position field should be non nullable
We have too many frontend issues linked to position field being empty, we shouldn't let this happen in the first place.
Let's make sure it's enforced at Postgres level

### LINKED ISSUE #8810: Tasks and Notes created from Record show page have a null position
## Bug Description

Tasks and Notes created from Record show page have a null position

## Expected behavior

1) they should be sent with 'last'
2) the backend should not allow a null for positions (we should add a constraint + make sure the backend applies the defaultValue

Also: backfill existing positions

### LINKED ISSUE #8971: Investigation sync-metadata performances

### PR DESCRIPTION
**Pull request by Weiko**
Fixes https://github.com/twentyhq/twenty/issues/8810
Fixes https://github.com/twentyhq/twenty/issues/5268
Fixes https://github.com/twentyhq/twenty/issues/8971

- Fixing Task/Note creation not sending position during creation
- Adding a command to backfill position being null, using existing backfill command. 
- Removed unused backfill job.
- Updated workspace entities to set position non-nullable and set a default value to make it non-required on the API
- Updated position factory to set a default position for all objects having a POSITION field instead of only company/people
- Moved the try/catch in each resolver factory calling GraphqlQueryRunnerException handler, makes more sense to call it in the actual graphql-query-runner and removing some duplicate codes
- Adding validations for input in QueryRunnerArgs factories
- Allow sync-metadata to override and sync defaultValues for certain field types (that can't be updated by users)
- Removing health-check from sync-metadata command during force mode to improve performances

### CODE REVIEW

File: packages/twenty-server/src/engine/api/graphql/graphql-query-runner/resolvers/graphql-query-create-one-resolver.service.ts [hunk 1]
```diff
@@ -83,8 +83,6 @@ export class GraphqlQueryCreateOneResolverService extends GraphqlQueryBaseResolv
     });
   }
 
-  async;
```
Comment (magrinj) [inline packages/twenty-server/src/engine/api/graphql/graphql-query-runner/resolvers/graphql-query-create-one-resolver.service.ts:86]: I remove this too inside my PR haha

File: packages/twenty-server/src/engine/api/graphql/workspace-query-runner/factories/query-runner-args.factory.ts [hunk 2]
```diff
@@ -39,9 +38,10 @@ export class QueryRunnerArgsFactory {
     const fieldMetadataMapByNameByName =
       options.objectMetadataItemWithFieldMaps.fieldsByName;
 
-    const shouldBackfillPosition = hasPositionField(
-      options.objectMetadataItemWithFieldMaps,
-    );
+    const shouldBackfillPosition =
+      options.objectMetadataItemWithFieldMaps.fields.some(
```
Comment (magrinj) [inline packages/twenty-server/src/engine/api/graphql/workspace-query-runner/factories/query-runner-args.factory.ts:42]: Why removing the utility function, this one was only used there ?

File: packages/twenty-server/src/engine/api/graphql/workspace-query-runner/services/record-position-backfill-service.ts [hunk 3]
```diff
@@ -29,15 +32,20 @@ export class RecordPositionBackfillService {
...
 
-    const objectMetadataEntities =
-      await this.objectMetadataService.findManyWithinWorkspace(workspaceId, {
-        where: { isSystem: false },
-      });
-
-    const objectMetadataWithPosition =
-      objectMetadataEntities.filter(hasPositionField);
+    const objectMetadataCollection = await this.objectMetadataRepository.find({
+      where: {
```
Comment (magrinj) [inline packages/twenty-server/src/engine/api/graphql/workspace-query-runner/services/record-position-backfill-service.ts:36]: Much better 😅

**Review by magrinj — COMMENTED** [review]
Some comments, but globally good to me

File: packages/twenty-front/src/modules/activities/hooks/useOpenCreateActivityDrawer.ts [hunk 4]
```diff
@@ -31,7 +31,7 @@ export const useOpenCreateActivityDrawer = ({
   const setHotkeyScope = useSetHotkeyScope();
 
   const { createOneRecord: createOneActivity } = useCreateOneRecord<
-    Task | Note
+    (Task | Note) & { position: 'first' | 'last' }
```
[the remark this answers is not in the document]
Comment (Weiko) [reply packages/twenty-front/src/modules/activities/hooks/useOpenCreateActivityDrawer.ts:34]: It's only required for creation so I didn't want to update everything 🤔

File: packages/twenty-server/src/engine/api/graphql/graphql-query-runner/interfaces/base-resolver-service.ts [hunk 5]
```diff
@@ -56,77 +57,81 @@ export abstract class GraphqlQueryBaseResolverService<
...
+        graphqlQueryParser,
+        graphqlQuerySelectedFieldsResult,
+      };
 
-    const selectedFields = graphqlFields(options.info);
+      const results = await this.resolve(graphqlQueryResolverExecutionArgs);
 
-    const graphqlQuerySelectedFieldsResult =
-      graphqlQueryParser.parseSelectedFields(
+      const resultWithGetters = await this.queryResultGettersFactory.create(
```
[the remark this answers is not in the document]
Comment (Weiko) [reply packages/twenty-server/src/engine/api/graphql/graphql-query-runner/interfaces/base-resolver-service.ts:113]: True

File: packages/twenty-server/src/engine/api/graphql/workspace-query-runner/factories/query-runner-args.factory.ts (hunk 2 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/api/graphql/workspace-query-runner/factories/query-runner-args.factory.ts:42]: Yes

**Pull request merged**